controllers/actions.go

```diff
@@ -0,0 +1,46 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Action is a single operation which will be executed by the controller.
+type Action interface {
+	Execute(context.Context) error
+}
+
+// CreateRuntimeObject is an Action which creates the supplied API object
+type CreateRuntimeObject struct {
+	client client.Client
+	obj    runtime.Object
+}
+
+func (o *CreateRuntimeObject) Execute(ctx context.Context) error {
+	if err := o.client.Create(ctx, o.obj); err != nil {
+		return fmt.Errorf("error %q while creating object ", err)
+	}
+	return nil
+}
+
+// PatchStatus is an Action which patches the status of original with any
+// changed status fields of new.
```

Comment:
I don't think there's much risk, since we are executing the action immediately after it's been constructed.
But yeah, I agree. And the constructor option is probably easiest.
I'll do that in a follow up.